Implements lobby page to update users in room

aarti156 opened the pull request.

aarti156 committed:
Implements sockets to update people in lobbyroom

Home page emits socket event if there is a gameId in local storage
Socket event triggers lobby page to re-render for other players in the game
Changes to be committed:
modified:   public/home/home.js
modified:   public/lobby/lobby.js
modified:   server/server.js

aarti156 committed:
Cleans up code in lobby view

Changes to be committed:
modified:   public/lobby/lobby.js

hiteshlala commented on server/server.js:
```diff
@@ -60,10 +61,26 @@ io.on('connection', function(socket){
+     var playersInGame = usersInGames[data.gameId];
+     for(var i = 0; i < playersInGame.length ; i++) {
+      if(playersInGame[i].facebookId === data.user.facebookId) {
+        playersInGame.splice(i, 1);
```
instead of splicing in the for loop, just save the index to splice after or use a flag to decide if we are going to push or not

hiteshlala commented on server/server.js:
```diff
@@ -60,10 +61,26 @@ io.on('connection', function(socket){
+  socket.on('a user left lobby', function(data) {
+    socket.join(data.gameId);
+    for(var j = 0; j < usersInGames[data.gameId].length; j++) {
+      if(usersInGames[data.gameId][j].facebookId === data.user.facebookId) {
```
Is usersInGames defined here? 

 I get the error:
TypeError: Cannot read property 'length' of undefined
    at Socket.<anonymous> (/Users/hiteshlala/HackReactor/thesis2/server/server.js:76:49)
    at emitOne (events.js:77:13)

hiteshlala commented:
This pull request breaks the app on my computer.  I think that there is a variable that is not defined.

aarti156 commented:
Hi hitesh,

usersInGames is a global variable in server file.
Did you try deleting ur local storage before testing?

I tried testing in my computer and it works fine.. May be i am missing
something. Will check again!

On Wed, Mar 23, 2016 at 9:07 PM, Hitesh Lala notifications@github.com
wrote:

> This pull request breaks the app on my computer. I think that there is a
> variable that is not defined.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/luminescent-recliners/pokemon-board-game/pull/124#issuecomment-200653005

aarti156 committed:
Cleans up code in server file

Changes to be committed:
modified:   public/lobby/lobby.js
modified:   server/server.js

hiteshlala commented:
I will try again this morning Hitesh Lala

```
  From: Arthi Palaniappan <notifications@github.com>
```

 To: luminescent-recliners/pokemon-board-game pokemon-board-game@noreply.github.com 
Cc: Hitesh Lala hash_lala@yahoo.com
 Sent: Wednesday, March 23, 2016 11:59 PM
 Subject: Re: [pokemon-board-game] Implements lobby page to update users in room (#124)

Hi hitesh,

usersInGames is a global variable in server file.
Did you try deleting ur local storage before testing?

I tried testing in my computer and it works fine.. May be i am missing
something. Will check again!

On Wed, Mar 23, 2016 at 9:07 PM, Hitesh Lala notifications@github.com
wrote:

> This pull request breaks the app on my computer. I think that there is a
> variable that is not defined.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/luminescent-recliners/pokemon-board-game/pull/124#issuecomment-200653005
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
